### PR TITLE
[dev2] Adding new fields to `Node` serialization

### PR DESCRIPTION
**Pull request by franramirez688**
Changelog: Feature: Adding `prev` to `Node.serialize()` method.
Changelog: Feature: Adding subsettings to `Settings.serialize()` method.
Closes: https://github.com/conan-io/conan/issues/11327
Docs: omit

### CODE REVIEW

File: conans/model/settings.py [hunk 1]
```diff
@@ -172,7 +172,8 @@ def __init__(self, definition=None, name="settings", parent_value="settings"):
                       for k, v in definition.items()}
 
     def serialize(self):
-        return {k: v.value for k, v in self._data.items()}
+        # Returning all the information available, including the sub-settings
+        return self.items()
```
Comment (memsharded) [inline conans/model/settings.py]: But this change is not returning a "string" object, but a ``SettingsItem`` one.
The idea is that ``serialize()`` methods return hierarchy of python primitives, strings, dicts, lists, numbers...

File: conans/model/settings.py [hunk 2]
```diff
@@ -172,7 +172,8 @@ def __init__(self, definition=None, name="settings", parent_value="settings"):
                       for k, v in definition.items()}
 
     def serialize(self):
-        return {k: v.value for k, v in self._data.items()}
+        # Returning all the information available, including the sub-settings
+        return {k: v for k, v in self.items()}
```
Comment (memsharded) [inline conans/model/settings.py]: This was done on purpose to not keep using the legacy bit of mess that is behind ``self.items()``, that calls ``self.fields()``,  ``self.values_list``.

**Review by memsharded — COMMENTED** [review]
Looks good

**Review by lasote — APPROVED** [review]

File: conans/model/settings.py (hunk 2 above)
Comment (franramirez688) [reply conans/model/settings.py]: Ok, let me have a look at it.

Comment (franramirez688) [reply conans/model/settings.py]: Refactored a little bit

**Review by memsharded — APPROVED** [review]

**Pull request merged**